src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/BuildDataResults.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.CodeAnalysis.CSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities.RoslynWorkspaceUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject.RoslynWorkspaceUtilities;
+
+public class BuildDataResults
```

Comment:
Does this need to be public?

For every type introduced, we should make it internal if it's simple to do so. For the ones where it's less clear (for example you're using them in unit tests), let's have a discussion when we meet.

Reply:
In order to accomplish this, I will need to make other classes 'internal'.

I can implement these changes once I gain a clearer understanding of which APIs the service will utilize

Reply:
It's better to make them internal until a clear need for them to be public emerges. 

The main public methods that the Try .NET web service will need to call will be the kernel commands and events, and this infrastructure should be an implementation detail.